src/components/Filter.tsx

```diff
@@ -0,0 +1,52 @@
+"use client";
+import React, { /*Component,*/ useState } from "react";
+import Image, { StaticImageData } from "next/image";
+
+const Filter = ({
+  filters,
+  message,
+  //onChange,
+}: {
+  filters: Array<StaticImageData>;
+  message: Array<string>;
+  //onChange: (filter: StaticImageData) => void;
+}) => {
+  const [selectedFilter, setSelectedFilter] = useState(filters[0]);
+  const handleClick = ({ filter }: { filter: StaticImageData }) => {
```

Comment:
this should just be const handleClick = ( filter: StaticImageData )

how you did it before was causing the undefined error due to the fact the its reading {filter} as destructing an object